src/Http/Routing.Abstractions/src/IRouteConstraint.cs

```diff
@@ -24,7 +24,7 @@ public interface IRouteConstraint : IParameterPolicy
         /// </param>
         /// <returns><c>true</c> if the URL parameter contains a valid value; otherwise, <c>false</c>.</returns>
         bool Match(
-            HttpContext httpContext,
+            HttpContext? httpContext,
```

Comment:
I found this strange. None of the build in constraints check for a non-null constraint, but it might catch people off guard.

Reply:
hmm, didn't we say that we wanted to put the things in here that matched our expectations? I don't expect a constraint to receive a null httpcontext any time soon.

Reply:
There's a couple of APIs where this code runs that the framework explicitly passes in a null context: https://github.com/dotnet/aspnetcore/blob/master/src/Http/Routing/src/DefaultLinkGenerator.cs#L115

Reply:
HttpContext is optional because there is demand for LinkGenerator to work outside of the context of a HTTP request.